frontend/src/components/software/SoftwareDetailsContainer.jsx

```diff
@@ -2,9 +2,60 @@ import React, {Component} from 'react';
 import SoftwareDetailsComponent from './SoftwareDetailsComponent';
 
 class SoftwareDetailsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.addCall = this.addCall.bind(this);
+    this.state = {
+      name: '',
+      key: '',
+      availableKeys: '',
+      duration: '',
+      mode: '',
```

Comment:
Nie potrzebujemy przechowywać tutaj `mode`, gdyż jest ono zawsze dostępne poprzez `this.props`.

Reply:
poprawione